libfabric.spec.in

```diff
@@ -8,10 +8,15 @@ Url: http://www.github.com/ofiwg/libfabric
 Source: http://www.openfabrics.org/downloads/fabrics/%{name}-%{version}.tar.bz2
 Prefix: ${_prefix}
 
-BuildRequires: libnl-devel
+%if 0%{?configopts:1}
+# skip BuildRequires for custom build
```

Comment:
> Not without maintaining a map of provider : package names.

Don't we essentially have this anyway?  Every provider needs to know what their build dependencies are anyway.

In any case, I'm not really qualified to judge between this approach versus Jeff's.  Jeff, does this look okay?